addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -180,33 +181,20 @@ def _get_attachment_number(self):
         for record in self:
             record.attachment_number = attach_data.get(record.id, 0)
 
-    @api.multi
-    def _read_group_stage_ids(self, domain, read_group_order=None, access_rights_uid=None):
-        access_rights_uid = access_rights_uid or self.env.uid
-        Stage = self.env['hr.recruitment.stage']
-        order = Stage._order
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, read_group_order):
+        order = stages._order
         # retrieve job_id from the context and write the domain: ids + contextual columns (job or default)
         job_id = self._context.get('default_job_id')
         search_domain = [('job_id', '=', False)]
         if job_id:
             search_domain = ['|'] + search_domain + [('job_id', '=', job_id)]
-        if self.ids:
-            search_domain = ['|'] + search_domain + [('id', 'in', self.ids)]
-
-        stage_ids = Stage._search(search_domain, order=order, access_rights_uid=access_rights_uid)
-        stages = Stage.sudo(access_rights_uid).browse(stage_ids)
-        result = stages.name_get()
-        # restore order of the search
-        result.sort(lambda x, y: cmp(stage_ids.index(x[0]), stage_ids.index(y[0])))
-
-        fold = {}
-        for stage in stages:
-            fold[stage.id] = stage.fold or False
-        return result, fold
-
-    _group_by_full = {
-        'stage_id': _read_group_stage_ids
-    }
+        if stages:
+            search_domain = ['|'] + search_domain + [('id', 'in', stages.ids)]
```

Comment:
should be an `expression.OR` really. Possibly with a normalisation of the `search_domain` beforehand just in case.